main/lsp/LSPTypecheckerCoordinator.cc

```diff
@@ -57,9 +55,33 @@ void LSPTypecheckerCoordinator::syncRun(function<void(LSPTypechecker &)> &&lambd
     }
 }
 
+void LSPTypecheckerCoordinator::initialize(unique_ptr<InitializedParams> params) {
+    // TODO: Make async when we land preemptible slow path.
+    syncRun([&typechecker = this->typechecker, &workers = workers, &params = params](auto &tcd) -> void {
```

Comment:
I'm confused by this (and other lambdas) that accept an `auto &tcd` that they don't appear to use.  (It looks like it's a `LSPTypecheckerDelegate`?)  Is this just an intermediate artifact of your refactoring that will be removed in a future diff, or is this how it will end up?

Reply:
IRLed; this'll change later when we make these things async. `tcd` is used in all other invocations of `syncRun` which previously expected a `LSPTypechecker` object.